lib/taurus/core/tango/tangofactory.py

```diff
@@ -286,17 +287,21 @@ def getDevice(self, dev_name, create_if_needed=True, **kw):
            :raise: (taurus.core.taurusexception.TaurusException) if the given
                    dev_name is invalid.
         """
-        d = self.tango_devs.get(dev_name)
-        if d is None:
-            d = self.tango_alias_devs.get(dev_name)
-        if d is not None:
-            return d
-
         validator = _Device.getNameValidator()
         groups = validator.getUriGroups(dev_name)
+
         if groups is None:
             raise TaurusException("Invalid Tango device name '%s'" % dev_name)
 
+        d = self.tango_devs.get(dev_name)
+        if d is None:
+            d = self.tango_alias_devs.get(dev_name)
+            if d is not None:
```

Comment:
Shouldn't this block be unindented?
(I just say it by comparison with the original code)